repository/git_testing.go

```diff
@@ -42,6 +53,95 @@ func CreateTestRepo(bare bool) *GitRepo {
 	return repo
 }
 
+// setupSigningKey creates a GPG key and sets up the local config so it's used.
+// The key id is set as "user.signingkey". For the key to be found, a `gpg`
+// wrapper which uses only a custom keyring is created and set as "gpg.program".
+// Finally "commit.gpgsign" is set to true so the signing takes place.
+func (repo *GitRepo) setupSigningKey() {
```

Comment:
Same thing, this should not be part of the normal `GitRepo`, if only for the semantic.